docs/vm-configuration.md

```diff
@@ -0,0 +1,265 @@
+
+
+Security issues
+---------------
+
+There are many security features in libvirt, however, they are primarily focused
+around protection from malicious or misbehaving guests. For example, sVirt will
+stop guests attacking each other, or the host. NWFilter will stop guests doing
+IP/MAC/ARP spoofing on the network.
+
+There is a fine grained access control mechanism for the host facing management
+API, however, the checks are performed against the (subject, object, action)
+tuple. So while you can block a user or application from being able to invoke
+the 'virDomainDefineXML' action, if you allow this API, you cannot then control
+what information is passed in its parameters. For example, when configuring
+disks in the domain XML, arbitrary host paths can be specified as the backing
+storage for the virtual disk. Libvirt will happily let /etc/passwd be specified
+as the disk backing store. When libvirt starts the QEMU instance, it will then
+tell sVirt to grant access to the /etc/passwd file.
+
+IOW, any application which can invoke the virDomainDefineXML method on the
+libvirtd daemon has a privilege level which is equivalent to that of libvirtd
+itself, which is root. Thus applications calling virDomainDefineXML (and many
+other APIs) which accept configuration input from untrusted users, to feed into
+an XML document, must be very careful in their data handling.
+
+
+Config states
+-------------
+
+Naively one might expect a virtual machine to have a single document describing
+its configuration. In practice, this is unreasonably restrictive.
+
+The configuration provided by a tenant or management application will typically
+not express every single frontend or backend setting. When the XML document is
+provided to libvirt, it will fill in any important missing pieces of information
+in an attempt to fully represent the virtual hardware configuration and backend
+integration. For example, if the tenant did not supply PCI addresses, libvirt
+will assign them prior to starting QEMU. Even when libvirt fills in missing
+data this is only ever libvirt's best attempt at a complete virtual hardware
+configuration. For example, for ISA serial ports, libvirt does not record the
+hardware I/O port associated with the serial port, since that is not currently
+a configurable option & thus can be assumed stable without being explicitly
+represented. A future libvirt or QEMU, may allow I/O port configuration, and
+thus when upgrading to newer libvirt, the XML document previously considered
+to be complete may have yet more defaults filled in.
+
+Applications that wish to be able to start the guest multiple times, while
+guaranteeing a stable hardware ABI, need to capture the augmented configuration
+and use it on subsequent boot attempts. Depending on which libvirt APIs are
+used, the augmented configuration can be obtained without starting the guest,
+or at the same time as starting a guest.
```

Comment:
There's plenty of stuff where QEMU sets defaults which have no exposure/representation in the libvirt XML file. As long as you don't change the machine type though, these defaults should be stable across QEMU versions.